File: Home Work 40/Home Work 40/SecondViewController.m
//
//  SecondViewController.m
//  Home Work 40
//
//  Created by Антон Нарижный on 10.09.22.
//

#import "SecondViewController.h"
#import "ThirdViewController.h"
#import "ViewControllerModel.h"

@interface SecondViewController ()

@property (nonatomic) IBOutlet UILabel *titleLabel;
@property (nonatomic) NSMutableString *name2;

@end

@implementation SecondViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.titleLabel.text = self.screenName;
    
    NSLog(self.SecondViewModel.name);
}

- (IBAction)didTapOpenNextScreen:(id)sender {
    UIStoryboard *str = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    ThirdViewController *controller = [str instantiateViewControllerWithIdentifier:@"ThirdViewController"];
    
    controller.screenName = self.screenName;
    controller.ThirdViewModel = self.SecondViewModel;
    
    [self presentViewController: controller animated: YES completion: nil];
}

@end

File: Home Work 7/Home Work 7/SecondViewController.swift
import UIKit

class SecondViewController: UIViewController {

    @IBOutlet var label2: UILabel!
    @IBOutlet var animalLable: UILabel!
    
    var string2: String = ""
    var animalStatus: String = ""
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        label2.text = string2
        animalLable.text = animalStatus
        
        self.title = "Второй экран"
        
        let thirdScreen = UIBarButtonItem(title: "Third Screen", style: .done, target: self, action: #selector(openThirdScreen))
        navigationItem.rightBarButtonItem = thirdScreen
    }

    @IBAction func didTapMe2(){
        let str: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
        let vc: ThirdViewController = str.instantiateViewController(withIdentifier: "ThirdViewController") as! ThirdViewController
        
        vc.string3 = string2
        
        vc.modalPresentationStyle = .fullScreen
        vc.modalTransitionStyle = .coverVertical
        self.present(vc, animated: true)
    }
    
    @objc func openThirdScreen(){
        let str: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
        let vc: ThirdViewController = str.instantiateViewController(withIdentifier: "ThirdViewController") as! ThirdViewController
        
        vc.string3 = string2
        
        vc.modalPresentationStyle = .fullScreen
        vc.modalTransitionStyle = .coverVertical
        self.navigationController?.pushViewController(vc, animated: true)
    }
    
}
